### ISSUE TITLE
OpenVidu Init Scripts

### ISSUE THREAD

**Issue by viktara** [OpenVidu/openvidu#225]
It would be nice to have OpenVidu init scripts.

Raising this issue and will provide an init script shortly.

**Comment by pabloFuente** [comment]
What do you mean OpenVidu "init sripts"?
OpenVidu is meant to be run in a development environment with Docker, just as explained in every tutorial, and in production environments can be launched automatically in Amazon Web Services ([here](https://openvidu.io/docs/deployment/deploying-aws/)) or manually installed and run in Ubuntu ([here](https://openvidu.io/docs/deployment/deploying-ubuntu/)). There are no more commands or scripts to run OpenVidu than the ones you can see in this last section.

**Comment by viktara** [comment]
What I mean is init.d / systemd configurations linux (we run OpenVidu on debian).

Currently you suggest starting most services using init scripts in
"7. Init services"

but for OpenVidu

"8. Init Openvidu Server JAR executable"

I think starting production services that way is a bit flaky!

**Comment by pabloFuente** [comment]
Steps 7 and 8 of [Deploying OpenVidu on Ubuntu](https://openvidu.io/docs/deployment/deploying-ubuntu/) are the minimum, simpler set of commands to launch OpenVidu. Of course, having a good and robust production setup requires the automation of these services, but this can be done in a number of different ways. We'd be entering devops territory here, not being a topic directly related to OpenVidu.

To sum up, any Linux service that you might install in a server could be treated the same way. In fact, we don't use systemd. In our own deployements we use Ansible and supervisor. If you are interested in how we do it, the repo is this: https://github.com/OpenVidu/openvidu-cloud-devops/tree/dev

**Comment by viktara** [comment]
Sorry, but I disagree a bit here.
There are significant usecases for Video Conferencing where users do not want to put their communication backend in AWS.
Now, If we wanted to package OpenVidu for distribution with Debian (planning to do that next) or any of it's derivatives then that would require init scripts.

Are you saying that you do not want to accept init scripts into Openvidu?

Are we better to make a fork and consider Linux services there?

**Comment by pabloFuente** [comment]
Okey, so you want to contribute an init script(s), that is a good choice for your set up but you think could also be used by other developers.
If it is that, then of course, there's no problem at all. Any help will be appreciated.

Keep us informed about your progress in this very same issue. I'll reopen it.

**Comment by pabloFuente** [comment]
Any update about this?

**Comment by viktara** [comment]
One of our team is working on the init scripts, let me get him to update here

**Comment by anantsaraswat** [comment]
@pabloFuente @viktara I have added our openvidu init script, Please check the attached.

[openvidu.txt](https://github.com/OpenVidu/openvidu/files/3107753/openvidu.txt)

**Comment by viktara** [comment]
Looks good to me.

@pabloFuente shall we have a directory for openvidu_packaging and put the init script and the rest the packaging stuff there?

**Comment by pabloFuente** [comment]
Hi,

> We're planning debian packages followed by rpm's

That's cool, but you will have to host your init scripts and rest of configuration files in your own repository.
I mean, the script looks good, but we cannot take care of a debian package of OpenVidu. Kurento Media Server is in fact one, but that's because it makes sense as a monolithic service.
OpenVidu, on the other hand, is a wrapper around KMS that is intended to be scalable and fault tolerance. Production common setups will be one OpenVidu Server and multiple KMS. That topology does not fit right now into a single debian package.

But of course you have complete freedom to implement and offer your own debian package if you want. Apache 2 License gives you that privilege.

Best regards.

**Comment by viktara** [comment]
That's cool, I'll create a repo for Linux packaging then.

Would be useful to know where else is a priority for contributions, we have some angular / ionic developers in our team too.

**Comment by pabloFuente** [comment]
Hi,

Thanks for offering contributions! Currently there's no need of Angular or Ionic contributions. We have some tutorials making use of these frameworks, but they are closed samples that do not require many changes. But we will ask for help if we need it in the future!

Thanks
